enhancements/cvo/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,134 @@
+---
+title: use-manifest-annotation-for-object-removal
+authors:
+  - "@jottofar"
+reviewers:
+  - "@LalatenduMohanty"
+  - "@wking"
+  - "@crawford"
+  - "@sdodson"
+approvers:
+  - "@crawford"
+  - "@sdodson"
+creation-date: 2020-08-03
+last-updated: 2020-08-04
+status: implementable
+---
+
+# Use Manifest Annotation For Object Removal
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently CVO managed object removal is handled by jobs.
+Jobs have many immutable components, so they aren't a great match for CVO managed manifests.
+This enhancement replaces that approach with a new manifest annotation requesting the CVO delete the in-cluster object instead of creating/reconciling it.
+This will provide a more straightforward way to remove content.
+
+## Motivation
+
+There should be a straightforward way to remove CVO managed objects.
+
+### Goals
+
+* Users will be able to remove any of the currently managed CVO objects by simply modifying the object's existing manifest and adding the new delete annotation.
+
+### Non-Goals
+
+* This enhancement does not add any new objects for managment by CVO, therefore the new delete annotation only applies to the currently supported manifests and any that may be added in the future.
+
+## Proposal
+
+When the following annotation appears in a CVO supported manifest and is set to "true" the associated object will be removed from the cluster by the CVO.
```

Comment:
Is there any deletion ordering guarantee?

Reply:
Same as execution order?  I would prefer not inventing a new order unless there's a use case reason.  Use case?

Reply:
But I do need to change my Story 2 as I did not account for deleting items in reverse order.

Reply:
> Same as execution order? I would prefer not inventing a new order unless there's a use case reason. Use case?

You usually want to delete things in the *opposite* order you create them.  Imagine the simple/common case of creating a namespace and then creating a deployment.  If you delete in the same order you create them, you can actually make it impossible to clean up resources.

Reply:
Agreed on that, and so then that implies that delete happens.... at the beginning?  At the end?

Reply:
Agree. That's explained in the changes to the enhancement. The order is defined by the dev just like create order they just have to reverse it.